### ISSUE TITLE
Create a new Channel

### ISSUE THREAD

**Issue by himanshulalarya** [hyperledger-labs/blockchain-automation-framework#1030]
### Description
Create a new channel under a running Fabric Network. v2.x
Run the BAF network. Create a new channel on top of it.

Additional Information:
The feature of creating new channel can also be developed and tested on fabric v2.2 which is currently under development in branch: feature/fabric220

### Acceptance Criteria
1. Stage I -  Create a new channel and add peers of existing organization to it.
2. The development should be done by talking the latest code from develop branch.

**Comment by sivaramsk** [comment]
I am interested in this ticket, I will look at the code to see whether I can take this up.

**Comment by sivaramsk** [comment]
I referred this document for this process - https://hyperledger-fabric.readthedocs.io/en/release-2.2/create_channel/create_channel.html

If I understood this correctly, following are the 
1. Download the crypto-materital for the orderer and the orgs that are part of the channel.
2. Create a new configtx.yaml with the new channel.
3. Create channel artifacts for the new channel profile.
4. Create the channel from the artifacts with the orderers cryptomaterials.
5. For each org that is part of the channel, join the channel
6. Update anchor peer.

Does the above steps makes sense? Also in the acceptance criteria, stage 1 says create new channel without defining any network properties. For creating a channel you atleast need orderer's crypto materials right?

**Comment by himanshulalarya** [comment]
Yes, As per stage one. I've updated the criteria. The developer needs to create a new channel and add the peers of existing organization to it in a running network.
Our BAF code is equipped to support with Multiple peers and channel. You can surely try and test it.

**Comment by sivaramsk** [comment]
I tested for adding two channels when creating the network and that does not work. The channel gets created though, I see the job getting run successfully and creating the block. But the job for the creator org to join the channel is not getting triggered. I taking a look why the job is not getting created, does it make sense to debug this??

**Comment by himanshulalarya** [comment]
@sivaramsk , Yes, I think this needs to looked upon. Since we haven't done a multi channel test before, there might be some configurational requirements missing. You may need to backtrace all the ansible roles, related to channel_creation for this.

**Comment by sivaramsk** [comment]
I made a mistake in the configuration while testing multiple channels during initial network deployment. When I configured the 2nd channel, I removed the key "org_status: new" as I saw deploy_network.yaml does not check the key. Once I added key again the channel get added properly in the initial deployment. 

One question I have is, when does consortium created? When I defined the 2nd channel with a new consortium the deployment failed and I have the 2nd channel in the same consortium as the 1st channel deployment succeeds. I plan to do more testing methodically to understand what works and what does not work, but having different consortium should also work right?

**Comment by jagpreetsinghsasan** [comment]
Currently, BAF doesn't support multiple Consortium, because bringing organizations from different consortium to ours is a task with so many limitations and conditions. We have to enable multiple channels within the same consortium for this issue.

**Comment by sivaramsk** [comment]
I confirm multi-channel works during network deployment and we just have to document the same. For now I have started looking in to adding a new channel to the existing network. 

I encountered a weird error while testing, I had the below channel configuration

Channel 1:
  - Org1
  - Org2
  - Org3
Channel 2:
  - Org1
  - Org2

The ansible took to create the "Channel 2" first and that went well and when the loop came to create "Channel 1", it threw and error saying Org 3 is not part of the consortium. I could not reproduce the issue after that one time, but guessed there might be some issue when adding a 2nd channel and a new org is present. I would do more testing on this later, but does this error strike anything for you guys?

**Comment by himanshulalarya** [comment]
Hi @srinivasansankaran "when adding a 2nd channel and a new org is present." have you added a new org and added it to channel1 before making  channel2 ?

**Comment by sivaramsk** [comment]
Sorry for confusing statements, hard for me to explain it more clearly. Let me try again, this was part of testing with calling "site.yaml" and install 2 channels. The configuration had 3 orgs, 2 channels as I mentioned in my earlier comment. 

1. All the 3 orgs got added to the network successfully. I can see the peer come up sucessfully.
2. When the channels are getting added, instead of "channel 1", "channel 2" got added first and the org1 and org2 joined the channel successfully. 
3. Channel 1 also got created successfully, but when the org 3 was about get added to the channel, it failed with the error saying it is not part of the consortium. 

As I said earlier, I could not reproduce the issue after the first time as I noticed Channel 1 got created 1st all times after that. Is the description of the error clear now?

**Comment by jagpreetsinghsasan** [comment]
The channel block creation roles should be changed to support multi-peer.
The consortium error came because the templates used for creating the configtx needs change (and maybe other roles as well, scoped in this issue). 
The templates used to create the configtx are located [here ](https://github.com/hyperledger-labs/blockchain-automation-framework/tree/develop/platforms/hyperledger-fabric/configuration/roles/create/configtx/templates)

**Comment by sivaramsk** [comment]
Team - I can confirm with minimal changes I was able to add a new channel to an existing cluster. I do some more testing before submitting a PR for review. 

I have to say this is pretty well written code. All I had to do was to create a playbook out of appropriate roles from deploy_network and test it. The initial comment by @himanshulalarya that the BAF roles are equipped to handle this feature is true.

**Comment by sivaramsk** [comment]
Hi - created a pull request. I just modified an existing playbook create-join-channel.yaml to read a new variable in the networky.yaml called "channel_status" and add the channel. I was not sure whether to modify the existing playbook itself.

As of now I have only tested adding a new channel, I have not tested installing chaincode in to the channel. Will test that and update the result as well.

**Comment by sivaramsk** [comment]
Team - I have tested installing a chaincode to newly added channel and it works as expected. I have also added a new md file with instructions to add a new channel. One question I have is, should the playbook be a separate playbook named add-new-channel.yaml or should I modify the existing file create-join-channel.yaml?
